types/interfaces.d.ts

```diff
@@ -71,8 +71,21 @@ export type IChartOptions<O = {}> = DeepPartial<
     O
 >;
 
+export enum ChartTypeEnum {
+  bar = 'bar',
+  bubble = 'bubble',
+  doughnut = 'doughnut',
+  line = 'line',
+  pie = 'pie',
+  polarArea = 'polarArea',
+  radar = 'radar',
+  scatter = 'scatter',
+}
+
+export type IChartType = keyof typeof ChartTypeEnum;
+
 export interface IChartConfiguration<
-  TYPE = string,
+  TYPE extends IChartType = IChartType,
```

Comment:
As @emmcbd mentioned, it should be possible to remove this line and use `IChartType` instead of `TYPE` at line 94.

Reply:
Yes, I got it. I just need to confirm something. It looks like it will affect the types of chart controllers.
`IBarControllerConfiguration`, `ILineControllerConfiguration`, etc. The type will not be specified explicitly. Are we ok with this?
@emmcbd what do you think?

Reply:
Maybe you need to use the Enum and not the keyof typeof of it here.

Reply:
Not sure I got it. What I meant was that if we remove the genetic type TYPE, then the chart controllers will not be able to specify the chart type explicitly.

Reply:
Ah right, I was reading the tempalte thing wrong. I don't think the TYPE should be removed.
Maybe the `extends IChartType` part of the definition can be removed? Anyway, does not probably make a difference.

Reply:
On second thought, after reading again the `I*ControllerConfiguration` type definition, I agree that TYPE should not be removed, it allows to affect the type to the corresponding controller.

Reply:
It makes a little different. The "extends" marks that the TYPE should match IChartType.
Let me show you: [example](https://www.typescriptlang.org/play?#code/C4TwDgpgBAwgFgQwE7ACrmgXigcgEbI5QA+uANgJYB2EOA3AFAMQAeYA9ilNcBEgGYIAxtACS8ZMBjsq-CgHMArkgTAKMgDyoAmgAUAolFa8qAEwDOsRCnSQo2CTYwA+KAG8GUL1FCQAXFA6BowAvsxsnMDcVLwCwmKOwAByMvosJhZaeoYO1mgu7p7evhABQfqhTKwcXCVQogBCyNIxSOxkZHwtckoqapqo9lBUigC2eHwANFAAMkPmwEjU8q7Y4nndCsqq6lQaODjOdFAA9CdQSQDy4TVRdaIAYuzsLYvtnUibvTsDQyPjU1m80Wy1W9USX22-T2OEoNEOxzOUEuAGkqhFahh6jNqBBXm0Ol0ZD0obstH8xhMkNM5tgFksqCshutJCkqGkMuZ9gjTudUTdIj4saIZjI8TI3oTPsStn0yYNsP8qTTgQymWtEmyORAzFzYbieUjUUA)
try to use autocomplete also.